src/xdp-app-info.c

```diff
@@ -1104,6 +1104,60 @@ xdp_invocation_ensure_app_info_sync (GDBusMethodInvocation  *invocation,
                                               cancellable,
                                               error);
 }
+static XdpAppInfo *
+transition_host_app_info_to_registered  (GDBusMethodInvocation  *invocation,
+                                         const char             *app_id,
+                                         GCancellable           *cancellable,
+                                         GError                **error)
+{
+  GDBusConnection *connection = g_dbus_method_invocation_get_connection (invocation);
+  const char *sender = g_dbus_method_invocation_get_sender (invocation);
+  g_autoptr(XdpAppInfo) registered_app_info = NULL;
+  XdpAppInfo *app_info;
+
+  g_assert (app_info_by_unique_name != NULL);
+
+  G_LOCK (app_infos);
+  app_info = g_hash_table_lookup (app_info_by_unique_name, sender);
+  G_UNLOCK (app_infos);
```

Comment:
Might want to hold the lock until the end of this function, and have a "already locked" version of `cache_insert_app_info()`.